pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1623,6 +1635,25 @@ func (tmpl *Template) GetType() TemplateType {
 	return TemplateTypeUnknown
 }
 
+// GetNodeType returns the type of this template as a Node Type
+//
+func (tmpl *Template) GetNodeType() NodeType {
+	switch tmpl.GetType() {
```

Comment:
tests?

Reply:
function already existed in `util`, so completely removed this fxn